References
============

Credits & Acknowledgements
+++++++++++++++++++++++++++

The pipeline is based on `nf-core's <https://pubmed.ncbi.nlm.nih.gov/32055031/>`_ `viralrecon <https://nf-co.re/viralrecon/1.1.0>`_ pipeline version 1.1.0. 

**nf-core**
    Ewels, P., Peltzer, A., Fillinger, S., Patel, H., Alneberg, J., Wilm, A., Garcia, M. U., Di Tommaso, P., & Nahnsen, S. (2020). The nf-core framework for community-curated bioinformatics pipelines. Nature biotechnology, 38(3), 276–278. `<https://doi.org/10.1038/s41587-020-0439-x>`_

Tools & Assets
+++++++++++++++++

**Nextflow**
    Di Tommaso, P., Chatzou, M., Floden, E. W., Barja, P. P., Palumbo, E., & Notredame, C. (2017). Nextflow enables reproducible computational workflows. Nature biotechnology, 35(4), 316–319. `<https://doi.org/10.1038/nbt.3820>`_. PubMed PMID: 28398311.

**FastQC**
    Andrews, S. (2019). A Quality Control tool for High Throughput Sequence Data. Babraham Bioinformatics. `<https://www.bioinformatics.babraham.ac.uk/projects/fastqc/>`_

**BCFtools**
    Li H. A statistical framework for SNP calling, mutation discovery, association mapping and population genetical parameter estimation from sequencing data. Bioinformatics. 2011 Nov 1;27(21):2987-93. doi: 10.1093/bioinformatics/btr509. Epub 2011 Sep 8. PubMed PMID: 21903627; PubMed Central PMCID: PMC3198575. `<https://www.ncbi.nlm.nih.gov/pubmed/21903627/>`_

**BEDTools**
    Quinlan AR, Hall IM. BEDTools: a flexible suite of utilities for comparing genomic features. Bioinformatics. 2010 Mar 15;26(6):841-2. doi: 10.1093/bioinformatics/btq033. Epub 2010 Jan 28. PubMed PMID: 20110278; PubMed Central PMCID: PMC2832824. `<https://www.ncbi.nlm.nih.gov/pubmed/20110278/>`_

**Bowtie 2**
    Langmead B, Salzberg SL. Fast gapped-read alignment with Bowtie 2. Nat Methods. 2012 Mar 4;9(4):357-9. doi: 10.1038/nmeth.1923. PubMed PMID: 22388286; PubMed Central PMCID: PMC3322381. `<https://www.ncbi.nlm.nih.gov/pubmed/22388286/>`_

**fastp**
    Chen S, Zhou Y, Chen Y, Gu J. fastp: an ultra-fast all-in-one FASTQ preprocessor. Bioinformatics. 2018 Sep 1;34(17):i884-i890. doi: 10.1093/bioinformatics/bty560. PubMed PMID: 30423086; PubMed Central PMCID: PMC6129281. `<https://www.ncbi.nlm.nih.gov/pubmed/30423086/>`_

**MultiQC**
    Ewels, P., & Magnusson, M. (2016). MultiQC: summarize analysis results for multiple tools and samples in a single report. Oxford Academic Bioinformatics, 32(19). `<https://doi.org/10.1093/bioinformatics/btw354>`_

**iVar**
    Grubaugh ND, Gangavarapu K, Quick J, Matteson NL, De Jesus JG, Main BJ, Tan AL, Paul LM, Brackney DE, Grewal S, Gurfield N, Van Rompay KKA, Isern S, Michael SF, Coffey LL, Loman NJ, Andersen KG. An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar. Genome Biol. 2019 Jan 8;20(1):8. doi: 10.1186/s13059-018-1618-7. PubMed PMID: 30621750; PubMed Central PMCID: PMC6325816. `<https://www.ncbi.nlm.nih.gov/pubmed/30621750/>`_

**Freyja** 

**Kraken 2**
    Wood DE, Lu J, Langmead B. Improved metagenomic analysis with Kraken 2. Genome Biol. 2019 Nov 28;20(1):257. doi: 10.1186/s13059-019-1891-0. PubMed PMID: 31779668; PubMed Central PMCID: PMC6883579. `<https://www.ncbi.nlm.nih.gov/pubmed/31779668/>`_

**Nextstrain**
    Hadfield J, Megill C, Bell SM, Huddleston J, Potter B, Callender C, Sagulenko P, Bedford T, Neher RA. Nextstrain: real-time tracking of pathogen evolution. Bioinformatics. 2018 Dec 1;34(23):4121-4123. doi: 10.1093/bioinformatics/bty407. PubMed PMID: 29790939; PubMed Central PMCID: PMC6247931. `<https://pubmed.ncbi.nlm.nih.gov/29790939/>`_

**pangolin**
    Áine O'Toole, Emily Scher, Anthony Underwood, Ben Jackson, Verity Hill, JT McCrone, Chris Ruis, Khali Abu-Dahab, Ben Taylor, Corin Yeats, Louis du Plessis, David Aanensen, Eddie Holmes, Oliver Pybus, Andrew Rambaut. pangolin: lineage assignment in an emerging pandemic as an epidemiological tool. Publication in preparation.

**picard-tools**
    `<https://broadinstitute.github.io/picard/>`_

**R**
    R Core Team (2017). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. `<https://www.r-project.org/>`_

**SAMtools**
    Li H, Handsaker B, Wysoker A, Fennell T, Ruan J, Homer N, Marth G, Abecasis G, Durbin R; 1000 Genome Project Data Processing Subgroup. The Sequence Alignment/Map format and SAMtools. Bioinformatics. 2009 Aug 15;25(16):2078-9. doi: 10.1093/bioinformatics/btp352. Epub 2009 Jun 8. PubMed PMID: 19505943; PubMed Central PMCID: PMC2723002. `<https://www.ncbi.nlm.nih.gov/pubmed/19505943/>`_

**SnpEff**
    Cingolani P, Platts A, Wang le L, Coon M, Nguyen T, Wang L, Land SJ, Lu X, Ruden DM. A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3. Fly (Austin). 2012 Apr-Jun;6(2):80-92. doi: 10.4161/fly.19695. PubMed PMID: 22728672; PubMed Central PMCID: PMC3679285. `<https://www.ncbi.nlm.nih.gov/pubmed/22728672/>`_

**SnpSift**
    Cingolani P, Patel VM, Coon M, Nguyen T, Land SJ, Ruden DM, Lu X. Using Drosophila melanogaster as a Model for Genotoxic Chemical Mutational Studies with a New Program, SnpSift. Front Genet. 2012 Mar 15;3:35. doi: 10.3389/fgene.2012.00035. eCollection 2012. PubMed PMID: 22435069; PubMed Central PMCID: PMC3304048. `<https://www.ncbi.nlm.nih.gov/pubmed/22435069/>`_